src/hooks/useXtraAssessment.tsx

```diff
@@ -35,7 +36,10 @@ function useXtraAssessment(connectionData: AssessmentConnectionData, onResponse:
   // https://github.com/Sumit1993/react-native-use-websocket#readme
   let default_options = {
     queryParams: queryParams, //queryParams, //{...props.connection.queryParams, queryParams}
-    onOpen: () => console.log(Date() + ' XtraServer-AI Connection opened'),
+    onOpen: () => {
+      console.log(Date() + ' XtraServer-AI Connection opened')
+      WebSocketOpenHandler({ requestedAt: queryParams.requested_at, authToken: queryParams.auth_token });
```

Comment:
@Jestin64 
Just pass all queryParams and use what you want to use in WebSocketOpenHandler